BTCPayServer/Controllers/UIServerController.Translations.cs

```
@@ -122,5 +124,70 @@ public async Task<IActionResult> DeleteDictionary(string dictionary)
             TempData[WellKnownTempData.SuccessMessage] = StringLocalizer["Dictionary {0} deleted", dictionary].Value;
             return RedirectToAction(nameof(ListDictionaries));
         }
+
+        [HttpPost("server/dictionaries/download")]
+        public async Task<IActionResult> DownloadLanguagePack(string language)
+        {
+            if (string.IsNullOrEmpty(language))
+            {
+                TempData[WellKnownTempData.ErrorMessage] = StringLocalizer["Please select a language"].Value;
+                return RedirectToAction(nameof(ListDictionaries));
+            }
+
+            try
+            {
+                var httpClient = HttpClientFactory.CreateClient();
+                var fileName = language.ToLowerInvariant();
+                var url = $"https://raw.githubusercontent.com/btcpayserver/btcpayserver-translator/main/translations/{fileName}.json";
+                var response = await httpClient.GetAsync(url);
+
+                if (!response.IsSuccessStatusCode)
+                {
+                    TempData[WellKnownTempData.ErrorMessage] = StringLocalizer["Failed to download language pack: {0}", response.StatusCode].Value;
+                    return RedirectToAction(nameof(ListDictionaries));
+                }
+
+                var translationsJson = await response.Content.ReadAsStringAsync();
+
+                if (string.IsNullOrEmpty(translationsJson) || translationsJson == "{}")
+                {
+                    TempData[WellKnownTempData.ErrorMessage] = StringLocalizer["Translation file is empty"].Value;
+                    return RedirectToAction(nameof(ListDictionaries));
+                }
+
+                if (!Translations.TryCreateFromJson(translationsJson, out var translationsObj))
```

Same, useless. Use `CreateFromJson`, let it crash if that's not right json.